src/providers/php.rs

```diff
@@ -0,0 +1,211 @@
+            if let Ok(Some(mut node_setup)) = nodejs.setup(app, env) {
+                pkgs.append(&mut node_setup.pkgs);
+            }
+        }
+
+        Ok(Some(SetupPhase::new(pkgs)))
+    }
+
+    fn install(
+        &self,
+        app: &crate::nixpacks::app::App,
+        env: &crate::nixpacks::environment::Environment,
+    ) -> anyhow::Result<Option<crate::nixpacks::phase::InstallPhase>> {
+        let mut cmd = String::new();
+        cmd.push_str("mkdir -p /var/log/nginx && mkdir -p /var/cache/nginx");
+        let nodejs = NodeProvider {};
+        if app.includes_file("composer.json") {
+            cmd.push_str(" && composer install");
+        };
+        if nodejs.detect(app, env)? {
+            if !cmd.is_empty() {
+                cmd.push_str(" && ");
+            }
+            cmd.push_str(
+                nodejs
+                    .install(app, env)?
+                    .as_ref()
+                    .unwrap()
+                    .cmd
+                    .as_ref()
+                    .unwrap()
+                    .as_str(),
+            );
+        }
+        Ok(Some(InstallPhase::new(cmd)))
+    }
+
+    fn build(
+        &self,
+        app: &crate::nixpacks::app::App,
+        _env: &crate::nixpacks::environment::Environment,
+    ) -> anyhow::Result<Option<crate::nixpacks::phase::BuildPhase>> {
+        if let Ok(true) = NodeProvider::has_script(app, "prod") {
+            return Ok(Some(BuildPhase::new(
+                NodeProvider::get_package_manager(app) + " run prod",
+            )));
+        }
+        Ok(None)
+    }
+
+    fn start(
+        &self,
+        app: &crate::nixpacks::app::App,
+        _env: &crate::nixpacks::environment::Environment,
+    ) -> anyhow::Result<Option<crate::nixpacks::phase::StartPhase>> {
+        Ok(Some(StartPhase::new(format!(
+            "([ -e /app/storage ] && chmod -R ugo+w /app/storage); perl {} {} /nginx.conf && (php-fpm -y {} & nginx -c /nginx.conf)",
+            app.asset_path("transform-config.pl"),
+            app.asset_path("nginx.template.conf"),
+            app.asset_path("php-fpm.conf"),
+        ))))
+    }
+
+    fn static_assets(
+        &self,
+        _app: &crate::nixpacks::app::App,
+        _env: &crate::nixpacks::environment::Environment,
+    ) -> anyhow::Result<Option<crate::nixpacks::app::StaticAssets>> {
+        Ok(Some(static_asset_list! {
+            "nginx.template.conf" => include_str!("php/nginx.template.conf"),
+            "transform-config.pl" => include_str!("php/transform-config.pl"),
+            "php-fpm.conf" => include_str!("php/php-fpm.conf")
+        }))
+    }
+
+    fn environment_variables(
+        &self,
+        app: &crate::nixpacks::app::App,
+        _env: &crate::nixpacks::environment::Environment,
+    ) -> anyhow::Result<Option<crate::nixpacks::environment::EnvironmentVariables>> {
+        let mut vars = EnvironmentVariables::new();
+        vars.insert(
+            "PHP_VERSION".to_string(),
```

Comment:
What is this variable needed for?

Reply:
I had been using this to find the FastCGI path, but I'm doing that differently now, so I'll remove it.